### PR TITLE
fix(collector): show correct nvme capacity

### LINKED ISSUE #466: [BUG] NVMe does not show capacity when using namespaces
**Describe the bug**
nvme0

**Expected behavior**
nvme0n

**Screenshots**
image

**Log Files**
If related to missing devices or SMART data, please run the `collector` in DEBUG mode, and attach the log file.
See [/docs/TROUBLESHOOTING_DEVICE_COLLECTOR.md](docs/TROUBLESHOOTING_DEVICE_COLLECTOR.md) for other troubleshooting tips. 

```
docker run -it --rm -p 8080:8080 \
-v `pwd`/config:/opt/scrutiny/config \
-v /run/udev:/run/udev:ro \
--cap-add SYS_RAWIO \
--device=/dev/sda \
--device=/dev/sdb \
-e DEBUG=true \
-e COLLECTOR_LOG_FILE=/opt/scrutiny/config/collector.log \
-e SCRUTINY_LOG_FILE=/opt/scrutiny/config/web.log \
--name scrutiny \
ghcr.io/analogj/scrutiny:master-omnibus

# in another terminal trigger the collector
docker exec scrutiny scrutiny-collector-metrics run
```

The log files will be available on your host in the `config` directory. Please attach them to this issue. 

Please also provide the output of `docker info`

### PR DESCRIPTION
**Pull request by uhthomas**
Some nvme devices do not report their capacity through the usual 'user_capacity' field, instead the total capacity is reported with the 'nvme_total_capacity' field.

Fixes: #466

### CODE REVIEW

File: collector/pkg/detect/detect_test.go
Comment (uhthomas) [inline collector/pkg/detect/detect_test.go:1]: Apologies, there are more changes than I would have liked in this file. I use [gofumpt](https://github.com/mvdan/gofumpt) which is a bit stricter with formatting.

File: webapp/backend/pkg/models/collector/smart.go [hunk 1]
```diff
@@ -239,7 +237,23 @@ type SmartInfo struct {
...
+	switch {
+	case s.NvmeTotalCapacity > 0:
+		return s.NvmeTotalCapacity
+	case s.UserCapacity.Bytes > 0:
+		return s.UserCapacity.Bytes
+	}
+	return 0
+}
+
+type UserCapacity struct {
```
Comment (uhthomas) [inline webapp/backend/pkg/models/collector/smart.go:251]: I separated this out for the unit test.

**Comment by AnalogJ** [comment]
hey @uhthomas this is fantastic work! thanks for the PR 🥳

File: collector/pkg/detect/detect_test.go
Comment (AnalogJ) [reply collector/pkg/detect/detect_test.go:1]: haha I was going to say. It's all good :)

**Pull request merged**